### PR TITLE
[Merged by Bors] - feat(RingTheory/RootsOfUnity/AlgebraicallyClosed): new file

### PR DESCRIPTION
**Pull request by MichaelStollBayreuth**
This adds a new file `RingTheory.RootsOfUnity.AlgebraicallyClosed` that contains an instance showing that algebraically closed fields have enough `n`th roots of unity when `n` is not divisible by the characteristic.

We add a new file because of the additional imports required in addition to `RingTheory.RootsOfUnity.EnoughRootsOfUnity`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/RootsOfUnity/EnoughRootsOfUnityInstances.lean
Comment (YaelDillies) [inline Mathlib/RingTheory/RootsOfUnity/EnoughRootsOfUnityInstances.lean:1]: This filename is a bit of a mouthful. What about `RingTheory.RootsOfUnity.AlgebraicallyClosedField` or `RingTheory.RootsOfUnity.AlgebraicallyClosed`?

Comment (MichaelStollBayreuth) [reply Mathlib/RingTheory/RootsOfUnity/EnoughRootsOfUnityInstances.lean:1]: Sounds good.

File: Mathlib/RingTheory/RootsOfUnity/AlgebraicallyClosed.lean [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+-/
+
+namespace IsAlgClosed
+
+/-- An algebraically closed field `F` satisfies `HasEnoughRootsOfUnity F n` for all `n`
+that are not divisible by the characteristic of `F`. -/
+instance hasEnoughRootsOfUnity (F : Type*) [Field F] [IsAlgClosed F] (n : ℕ) [i : NeZero (n : F)] :
+    HasEnoughRootsOfUnity F n where
+  prim := by
+    have : NeZero n := NeZero.of_neZero_natCast F
```
Comment (YaelDillies) [inline Mathlib/RingTheory/RootsOfUnity/AlgebraicallyClosed.lean:30]:
```suggestion
    have : NeZero n := .of_neZero_natCast F
```
same below

**Review by YaelDillies — APPROVED** [review]
maintainer delegate

apart from that. Thanks!

File: Mathlib/RingTheory/RootsOfUnity/AlgebraicallyClosed.lean (hunk 1 above)
Comment (MichaelStollBayreuth) [reply Mathlib/RingTheory/RootsOfUnity/AlgebraicallyClosed.lean]: OK; done.

**Comment by MichaelStollBayreuth** [comment]
@YaelDillies Your "maintainer delegate" didn't seem to have an effect.

File: Mathlib/RingTheory/RootsOfUnity/AlgebraicallyClosed.lean [hunk 2]
```diff
@@ -0,0 +1,37 @@
+/-
+Copyright (c) 2024 Michael Stoll. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michael Stoll
+-/
+import Mathlib.RingTheory.RootsOfUnity.EnoughRootsOfUnity
+import Mathlib.NumberTheory.Cyclotomic.Basic
+
+/-!
+# Instances for HasEnoughRootsOfUnity
```
Comment (YaelDillies) [inline Mathlib/RingTheory/RootsOfUnity/AlgebraicallyClosed.lean:10]:
```suggestion
# Algebraically closed fields have enough roots of unity
```

Comment (MichaelStollBayreuth) [reply Mathlib/RingTheory/RootsOfUnity/AlgebraicallyClosed.lean:10]: I'd rather leave the title in view of eventually adding an instance for `Circle` as well.

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**